### DISCUSSION TITLE
Pictures (png files) within python-tkinter scripts

### DISCUSSION THREAD

**Discussion by Lautarob1** [pyinstaller/pyinstaller#8041] [Help]
How can I assure that 3 png files that are part of a python script (tkinter GUI) get actually emmbebbed so they can be seen by the exe file as the python script does?
I have used this command: pyinstaller --add-data "*.png:." Z:\Users\myself\tests\TestGUI_09.py
The 3 png files get in the path D:\Develp\Python\Pytest_venv\dist\TestGUI_09\_internal\LogoApp02.png 
and the exe file get as D:\Develp\Python\Pytest_venv\dist\TestGUI_09\TestGUI_09.exe

Now, when running the exe, it thrown the following error: 

 File "tkinter\__init__.py", line 4125, in __init__
  File "tkinter\__init__.py", line 4072, in __init__
_tkinter.TclError: couldn't open "LogoApp02.png": no such file or directory
[45036] Failed to execute script 'TestGUI_09' due to unhandled exception!

As posed, this is something that doesn't make any sense to me. if the exe file cannot fine LogoApp02.png in its own self generated structure, where can I place the png file, so it can be seen by the exe??

Any help will be highly appreciated,

**Comment by rokm** [comment]
You shouldn't be loading your resources using relative paths, since that makes your code dependent on current working directory. 

Instead, anchor your path to parent of `__file__`, e.g., `os.path.join(os.path.dirname(__file__), 'LogoApp02.png')`, as described in [the documentation](https://pyinstaller.org/en/stable/runtime-information.html#using-file).
